Triangulation_3/include/CGAL/Regular_triangulation_3.h

```diff
@@ -1324,14 +1324,48 @@ namespace CGAL {
 #endif
 
       Vertex_handle &hint = m_tls_hint.local();
+
       for( size_t i_point = r.begin() ; i_point != r.end() ; ++i_point)
       {
         bool success = false;
         const Weighted_point &p = m_points[i_point];
         while(!success)
         {
-          if (m_rt.try_lock_vertex(hint) && m_rt.try_lock_point(p))
+          // The 'hint' is unsafe to use immediately because we are in a regular triangulation,
+          // and the insertion of a (weighted) point in another thread might have hidden (deleted)
+          // the hint.
+#define CGAL_RT3_IGNORE_TDS_CONCEPT
+#ifdef CGAL_RT3_IGNORE_TDS_CONCEPT
+          if(!m_rt.tds().vertices().is_used(hint))
+#else
+          if(!m_rt.tds().is_vertex(hint))
+#endif
+          {
+            hint = m_rt.finite_vertices_begin();
+            continue;
+          }
+
+          // We need to make sure that while are locking the position P1 := hint->point(), 'hint'
+          // does not get its position changed to P2 != P1.
+          const Weighted_point hint_point_mem = hint->point();
```

Comment:
The `point()` method is not thread-safe. You should store `hint_point_mem` the same way you store `hint`: only refresh it once the lock is obtained.

Reply:
I think this whole `hide_point_mem` shenanigan is actually just useless. In other contexts, we need to ensure that the position of the lock is valid by checking the erase counter. Here, I just need to ensure that I locked _some_ valid place, and it doesn't matter if it's not where I thought it was.

Reply:
I agree, but the value of `hint_point_mem` cannot be trusted, unless you assign it once in the `try_lock` block.

Reply:
Sure, but I don't care about its value: of course it would ideally be where I think it is, but if it is not and it locked somewhere else, then the `locate` will just start from another position and that's alright as the hint might be wrong anyway in the first place. 

All I care about is that the `vh->cell()` is safe to use, not whether it is at an interesting position.

Reply:
Documenting it is difficult because we already don't expose the vertex/cell containers. Also, there are indeed some calls to undocumented functions from the TDS at a few points, but nothing "as bad" as calling the containers.

aa1e0acbeef802a1ae8cbdfeaec2f0808211d4d5 does something legal and the time loss compared to directly calling the function is hardly noticeable. Is this ok?

Reply:
I really do not like specializations, because that breaks easily. We have seen that on several occasions (for example the first version of the implementation of the structural filtering was almost only used in the test cases, but not in real uses cases).

Could it be related to the detection of a tag, instead?

Reply:
How would a tag ensure that other TDS without such tag be properly tested or used in real cases ? In any case, we don't have any tests checking that T3 concepts match the actual code requirements (and they don't).

Reply:
Anyway, nothing will ensure that "other TDS" will be properly tested, because there exists only one concept `TDS_3` in the world.

But the use of a tag (instead of a specialization) will ensure that the trick will not break if:
  - one derives from a TDS, or if
  - one adds new templates parameters to TDS_3, will default values,
  - other maybe other patterns that can break easily with a specialization, and that I do not know of.

The truth is just that specialization on full classes is generally bad, whereas specialization on tags is acceptable.

And the worst is that the trick could turn unused without any compiler warning or failure. That is exactly what happened with the structural filtering, that was triggered by a full-specialization on `CGAL::Epick`